src/nagini_translation/main.py

```diff
@@ -204,6 +204,10 @@ def main() -> None:
         '--sif',
         action='store_true',
         help='Verify secure information flow')
+    parser.add_argument(
+        '--arp',
+        action='store_true',
+        help='Use Abstract Read Permissions')
```

Comment:
So if I don't set this flag and use ARP, what happens?

Reply:
Silver reported an error because rd is not used, and then Nagini crashed. Changed it to report an unsupported error in this case. For this I added another context field and check this before generating ARP function calls.

Reply:
Great, thanks!